### PR TITLE
Feature/lp74 out lined text field unit

### PR DESCRIPTION
**Pull request by AndresIvanMedinaH**
**Task’s name**
OutlinedTextField - TextNumber

    **Task’s link to navigate to jira**
https://devaptivist.atlassian.net/browse/LP-74?atlOrigin=eyJpIjoiNjFjNjk5YmMwMjM4NDVkNThkYWE5MTcxOWNkOTI1MzQiLCJwIjoiaiJ9

    **General task’s description of what was added, deleted or updated in the code.**
As a developer, I must create a component like the image above

showing an OutlinedTextField, that implements a label in the top left corner and has the label text = unit

this TextField will only accept Int numbers

this TextField belongs to the screen Repair request

    **Attachment file of how looks your code in the app or in the preview if it is necessary**

OutlinedTextFieldNumber

### CODE REVIEW

**Review by IngKevel — APPROVED** [review]

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+                ),
+            textStyle = MaterialTheme.typography.labelMedium.copy(
+                color = MaterialTheme.colorScheme.onPrimary
+            ),
+            singleLine = true,
+            maxLines = 1,
+            shape = MaterialTheme.shapes.small,
+            keyboardOptions = KeyboardOptions.Default.copy(
+                keyboardType = KeyboardType.Email // Especifica el teclado para el campo de texto como el de correo electrónico
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt:52]: Delete the comment, in the case you add a comment do it in english

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            )
+
+        )
+    }
+}
+
+fun String.isValidEmail(): Boolean {
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt:59]: Delete and replace by a high order function

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutlinedTextFieldUnit.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            shape = MaterialTheme.shapes.small,
+            keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number)
+        )
+    }
+}
+
+
+
+
+
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutlinedTextFieldUnit.kt]: Delete these empty lines

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    return android.util.Patterns.EMAIL_ADDRESS.matcher(this).matches()
+}
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt:61]: Delete this extend function

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ) {
+        var textFieldValue by remember { mutableStateOf(TextFieldValue("")) }
+
+        OutlinedTextField(
+            value = textFieldValue,
+            onValueChange = {
+                textFieldValue = it
+                val isValid = it.text.isValidEmail()
+                onEmailValidated(isValid)
+            },
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutLinedTextFieldMail.kt:35]: Replace with a high order function

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/OutlinedTextFieldUnit.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        var textFieldValue by remember { mutableStateOf(TextFieldValue("")) }
+        val pattern = remember { Regex("^\\d+\$") }
+        val isInputError = false
+        OutlinedTextField(
+            value = textFieldValue,
+            onValueChange = {
+                if (it.text.isEmpty() || it.text.matches(pattern)) {
+                    textFieldValue = it
+                }
+            },
```
Comment (juanislas77) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/OutlinedTextFieldUnit.kt]: Replace with a high order function

**Review by juanislas77 — CHANGES_REQUESTED** [review]
Resolve the comments

**Review by juanislas77 — APPROVED** [review]
Nice job

**Pull request merged**